cmake/kokkos_arch.cmake

```
@@ -805,7 +805,8 @@ ENDIF()
 IF (KOKKOS_ENABLE_OPENMP)
   IF(KOKKOS_CLANG_IS_CRAY)
     COMPILER_SPECIFIC_LINK_OPTIONS(
-      Clang -fopenmp
+        CrayClang -fopenmp # Needed for cmake>=3.28
+        Clang -fopenmp
```

The alternative would be
```suggestion
        Default -fopenmp
```
since we have already established that we have a Cray compiler.

---

Thats fine by me if there is no objection. I can add a comment with the details.

---

I mean its somewhat weird since its inside a CLANG_IS_CRAY thing. I think I prefer the more verbose one. Makes the intent clearer.